src/components/UserSettings/GlobalSettings.js

```diff
@@ -145,7 +145,8 @@ class GlobalSettings extends Component {
 
   buildSections (onChange, translatedLabels) {
     const { draftValues } = this.state
-    const { config, msg } = this.props
+    const { config, msg, currentValues: { persistLocale, language } } = this.props
```

Comment:
IMHO `persistLocale` should be read (only) from `draftValues` to allow updating both language and persist  in one save.